src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateIO.cs

```diff
@@ -181,12 +207,234 @@ public void UpdatePackageReference(PackageSpec updatedPackageSpec, IPackageUpdat
         }
     }
 
+    /// <inheritdoc cref="IPackageUpdateIO.GetLatestVersionAsync(string, bool, ILogger, CancellationToken)"/>
+    public async Task<NuGetVersion?> GetLatestVersionAsync(
+        string packageId,
+        bool includePrerelease,
+        ILogger logger,
+        CancellationToken cancellationToken)
+    {
+        var sources = GetSourcesForPackage(packageId);
+        var lookups = new Task<NuGetVersion?>[sources.Count];
+        for (int source = 0; source < sources.Count; source++)
+        {
+            SourceRepository sourceRepository = sources[source];
+            // If package source is a local folder feed, it might not actually be async
+            lookups[source] = Task.Run(() => FindHighestPackageVersionAsync(sourceRepository, packageId, includePrerelease, logger, cancellationToken));
+        }
+
+        await Task.WhenAll(lookups);
+
+        NuGetVersion? highestVersion = null;
+        foreach (var task in lookups)
+        {
+            if (task.Result != null)
+            {
+                if (highestVersion == null || task.Result > highestVersion)
+                {
+                    highestVersion = task.Result;
+                }
+            }
+        }
+
+        return highestVersion;
```

Comment:
What is the error handling like if highestVersion is null? Will the tool exit gracefully? What is the error shown to the user?